service/worker/batcher/workflow.go

```diff
@@ -185,7 +176,7 @@ type (
 		// UpdateActivitiesOptionsParams is params only for BatchTypeUpdateActivitiesOptions
 		UpdateActivitiesOptionsParams UpdateActivitiesOptionsParams
 		// ResetActivitiesParams is params only for BatchTypeResetActivities
-		ResetActivitiesParams ResetActivitiesParams
+		ResetActivitiesParams *workflowservicepb.StartBatchOperationRequest_ResetActivitiesOperation
```

Comment:
you can't put a proto in a struct and have it serialized correctly by the dataconverter, it'll choose json and serialize the proto using encoding/json, but it needs to use protojson. the workflow argument needs to be proto from the top. see the hack in ResetParams.